lib_static/open_project/authentication/strategies/warden/jwt_oidc.rb

```diff
@@ -0,0 +1,82 @@
+module OpenProject
+  module Authentication
+    module Strategies
+      module Warden
+        class JwtOidc < ::Warden::Strategies::Base
+          include FailWithHeader
+
+          SUPPORTED_ALG = %w[
+            RS256
+            RS384
+            RS512
+          ].freeze
+
+          def valid?
+            @access_token = ::Doorkeeper::OAuth::Token.from_bearer_authorization(
+              ::Doorkeeper::Grape::AuthorizationDecorator.new(request)
+            )
+            @access_token.present? &&
+              begin
+                @unverified_payload, @unverified_header = JWT.decode(@access_token, nil, false)
+                @unverified_header.present? && @unverified_payload.present?
+              rescue JWT::DecodeError
+                false
+              end
+          end
+
+          def authenticate!
```

Comment:
🟡 This method is... long. 🤣 And there's a ton of nesting going on, maybe some guard clauses can probably help.

Quick question: what should be returned here? We have a lot of `return`s, which basically return `nil`, is it intended? Should `nil` be the return value?

Reply:
I believe the method is called here:  https://github.com/wardencommunity/warden/blob/master/lib/warden/strategies/base.rb#L55C9-L55C22
So, no return value is expected. `returns` are just to stop following method execution.

Reply:
Cool then. :D